zipline/pipeline/loaders/events.py

```diff
@@ -0,0 +1,203 @@
+        the dates that announcements were made, and we assume we knew about the
+        announcement on all prior dates.  This mode is only supported if
+        ``infer_timestamp`` is explicitly passed as a truthy value.
+        Dict mapping sids to DataFrames, Series, or DatetimeIndexes.
+
+        If the value is a DataFrame, it then represents dates on which events
+        occurred along with other associated values. If the DataFrame
+        contains a "timestamp" column, that column is interpreted as the date
+        on which we learned about the event. If the DataFrames do not contain a
+         "timestamp" column, we assume we knew about the event on all prior
+         dates.  This mode is only supported if ``infer_timestamp`` is
+         explicitly passed as a truthy value.
+
+    infer_timestamps : bool, optional
+        Whether to allow omitting the "timestamp" column.
+    dataset : DataSet
+        The DataSet object for which this loader loads data.
+
+    """
+
+    @abc.abstractproperty
+    def expected_cols(self):
+        raise NotImplemented('expected_cols')
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=None):
+        self.all_dates = all_dates
+        # Do not modify the original in place, since it may be used for other
+        #  purposes.
+        self.events_by_sid = (
+            events_by_sid.copy()
+        )
+        dates = self.all_dates.values
+
+        for k, v in iteritems(events_by_sid):
+            # Already a DataFrame
+            if isinstance(v, pd.DataFrame):
+                if TS_FIELD_NAME not in v.columns:
+                    if not infer_timestamps:
+                        raise ValueError(
+                            DF_NO_TS_NOT_INFER_TS_ERROR.format(
+                                timestamp_column_name=TS_FIELD_NAME,
+                                sid=k
+                            )
+                        )
+                    self.events_by_sid[k] = v = v.copy()
+                    v.index = [dates[0]] * len(v)
+                else:
+                    self.events_by_sid[k] = v.set_index(TS_FIELD_NAME)
+                # Once data is in a DF, make sure columns are correct.
+                cols_except_ts = (set(v.columns) -
+                                  {TS_FIELD_NAME})
+
+                # Check that all columns other than timestamp are as expected.
+                if cols_except_ts != self.expected_cols:
+                    raise ValueError(
+                        WRONG_COLS_ERROR .format(
+                            expected_columns=self.expected_cols,
+                            sid=k,
+                            resulting_columns=v.columns.values
+                        )
+                    )
+            # Not a DataFrame and we only expect 1 column
+            elif len(self.expected_cols) == 1:
```

Comment:
We should invert this check to flatten the logic here. This can be: `if len(self.expected_cols) != 1: raise ...` and then the rest of the code does not need to be indented.